service/src/main/java/com/epam/esm/service/TagService.java

```diff
@@ -59,6 +78,7 @@ public TagDto findById(long id) {
      * @throws EntityAlreadyExistsException in case when tag with specified name already exists
      * @return {@link TagDto} object that represents created tag
      */
+    @Transactional
```

Comment:
Why is it transactional?

Reply:
`persist()` call in underlying repository requires transactional context. Otherwise
![image](https://user-images.githubusercontent.com/42818718/137087643-f1f510a1-c7e2-49e4-9138-c2755c881699.png)

Reply:
Documentation says next:
The @PersistenceContext annotation has an optional attribute type, which defaults to PersistenceContextType.TRANSACTION. This default is what you need to receive a shared EntityManager proxy. The alternative, PersistenceContextType.EXTENDED, is a completely different affair: This results in a so-called extended EntityManager, which is not thread-safe and hence must not be used in a concurrently accessed component such as a Spring-managed singleton bean. Extended EntityManagers are only supposed to be used in stateful components that, for example, reside in a session, with the lifecycle of the EntityManager not tied to a current transaction but rather being completely up to the application.